platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceBean.java

```diff
@@ -362,96 +366,84 @@ public void removeApplication(String appName) {
   }
 
   /** {@inheritDoc}. */
-  @SuppressWarnings("unchecked")
   @Override
   public List<Map<String, Object>> getServices(String applicationID) {
-
-    List<Map<String, Object>> returnValues = new ArrayList<>();
-
     List<Service> services =
         configAdmin.listServices(getDefaultFactoryLdapFilter(), getDefaultLdapFilter());
+
     if (services.isEmpty()) {
-      return returnValues;
+      return Collections.emptyList();
     }
 
     Set<BundleInfo> bundleInfos = getBundleInfosForApplication(applicationID);
     if (bundleInfos == null) {
-      return returnValues;
+      return Collections.emptyList();
     }
 
-    Set<String> bundleLocations = new HashSet<>();
-    for (BundleInfo each : bundleInfos) {
-      bundleLocations.add(each.getLocation());
-    }
+    Set<String> bundleLocations =
+        bundleInfos.stream().map(BundleInfo::getLocation).collect(Collectors.toSet());
 
     MetaTypeService metatypeService = getMetaTypeService();
-    Set<MetaTypeInformation> metatypeInformation = new HashSet<>();
-    if (metatypeService != null) {
-      for (Bundle each : getContext().getBundles()) {
-        for (BundleInfo bundleInfo : bundleInfos) {
-          if (bundleInfo.getLocation().equals(each.getLocation())) {
-            metatypeInformation.add(metatypeService.getMetaTypeInformation(each));
-          }
-        }
-      }
-    }
-
-    for (Map<String, Object> each : services) {
-      if (each.containsKey("configurations")) {
-        List<Map<String, Object>> configurations =
-            (List<Map<String, Object>>) each.get("configurations");
-        for (Map<String, Object> item : configurations) {
-          if (item.containsKey("bundle_location")) {
-            String bundleLocation = (String) item.get("bundle_location");
-            if (bundleLocations.contains(bundleLocation)) {
-              returnValues.add(each);
-              break;
-            }
-          }
-        }
-      } else {
-        if (checkForMetaTypesForService(metatypeInformation, each)) {
-          returnValues.add(each);
-        }
-      }
-    }
+    Set<MetaTypeInformation> metatypeInformation =
+        (metatypeService == null)
+            ? Collections.emptySet()
```

Comment:
❗️ This is semantically different. In the original code, if this block was skipped because `metatypeService` was null, there was still a chance for `returnValues` to be populated (see line 407 of the original code) and then returned.